satpy/etc/readers/abi_l2_nc.yaml

```diff
@@ -333,184 +333,226 @@ datasets:
 
   abi_l2_cod:
-        file_reader: !!python/name:satpy.readers.abi_l2_nc.NC_ABI_L2
-        file_patterns:
-        # F (Full Disk) or C (CONUS)
-        - '{system_environment:2s}_{mission_id:3s}-L2-COD{area_code:1s}-{scan_mode:2s}_{platform_shortname:3s}_s{start_time:%Y%j%H%M%S%f}_e{end_time:%Y%j%H%M%S%f}_c{creation_time:%Y%j%H%M%S%f}.nc'
-        # M1 or M2 for mesoscale
-        - '{system_environment:2s}_{mission_id:3s}-L2-CODM{area_code:1d}-{scan_mode:2s}_{platform_shortname:3s}_s{start_time:%Y%j%H%M%S%f}_e{end_time:%Y%j%H%M%S%f}_c{creation_time:%Y%j%H%M%S%f}.nc'
+    file_reader: !!python/name:satpy.readers.abi_l2_nc.NC_ABI_L2
+    file_patterns:
+    # F (Full Disk) or C (CONUS)
+    - '{system_environment:2s}_{mission_id:3s}-L2-COD{scene_abbr:1s}-{scan_mode:2s}_{platform_shortname:3s}_s{start_time:%Y%j%H%M%S%f}_e{end_time:%Y%j%H%M%S%f}_c{creation_time:%Y%j%H%M%S%f}.nc'
+    # M1 or M2 for mesoscale
+    - '{system_environment:2s}_{mission_id:3s}-L2-COD{scene_abbr:2s}-{scan_mode:2s}_{platform_shortname:3s}_s{start_time:%Y%j%H%M%S%f}_e{end_time:%Y%j%H%M%S%f}_c{creation_time:%Y%j%H%M%S%f}.nc'
+    observation_type: "COD"
 
   # CSPP Geo keeps Day and Night algorithm outputs separate
   abi_l2_codd:
     file_reader: !!python/name:satpy.readers.abi_l2_nc.NC_ABI_L2
     file_patterns:
       - '{system_environment:2s}_{mission_id:3s}-L2-CODD{scene_abbr:s}-{scan_mode:2s}_{platform_shortname:3s}_s{start_time:%Y%j%H%M%S%f}_e{end_time:%Y%j%H%M%S%f}_c{creation_time:%Y%j%H%M%S%f}.nc'
+    observation_type: "CODD"
 
   abi_l2_codn:
     file_reader: !!python/name:satpy.readers.abi_l2_nc.NC_ABI_L2
     file_patterns:
       - '{system_environment:2s}_{mission_id:3s}-L2-CODN{scene_abbr:s}-{scan_mode:2s}_{platform_shortname:3s}_s{start_time:%Y%j%H%M%S%f}_e{end_time:%Y%j%H%M%S%f}_c{creation_time:%Y%j%H%M%S%f}.nc'
+    observation_type: "CODN"
 
   abi_l2_cps:
-        file_reader: !!python/name:satpy.readers.abi_l2_nc.NC_ABI_L2
-        file_patterns:
-          # F (Full Disk) or C (CONUS)
-        - '{system_environment:2s}_{mission_id:3s}-L2-CPS{area_code:1s}-{scan_mode:2s}_{platform_shortname:3s}_s{start_time:%Y%j%H%M%S%f}_e{end_time:%Y%j%H%M%S%f}_c{creation_time:%Y%j%H%M%S%f}.nc'
-          # M1 or M2 for mesoscale
-        - '{system_environment:2s}_{mission_id:3s}-L2-CPSM{area_code:1d}-{scan_mode:2s}_{platform_shortname:3s}_s{start_time:%Y%j%H%M%S%f}_e{end_time:%Y%j%H%M%S%f}_c{creation_time:%Y%j%H%M%S%f}.nc'
+    file_reader: !!python/name:satpy.readers.abi_l2_nc.NC_ABI_L2
+    file_patterns:
+      # F (Full Disk) or C (CONUS)
+    - '{system_environment:2s}_{mission_id:3s}-L2-CPS{scene_abbr:1s}-{scan_mode:2s}_{platform_shortname:3s}_s{start_time:%Y%j%H%M%S%f}_e{end_time:%Y%j%H%M%S%f}_c{creation_time:%Y%j%H%M%S%f}.nc'
+      # M1 or M2 for mesoscale
+    - '{system_environment:2s}_{mission_id:3s}-L2-CPS{scene_abbr:2s}-{scan_mode:2s}_{platform_shortname:3s}_s{start_time:%Y%j%H%M%S%f}_e{end_time:%Y%j%H%M%S%f}_c{creation_time:%Y%j%H%M%S%f}.nc'
```

Comment:
@yufeizhu600 I suppose with sections like this, now that `scene_abbr` handles both F/C and M1/M2 if I use a `:s`, I could just combine these into one pattern, right? I don't think there is anything in the reader that depends on these being two separate patterns.

Reply:
Yes, I think that should work.